hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/CustomAvroKeyGenerator.java

```diff
@@ -76,8 +76,11 @@ private static List<BaseKeyGenerator> getPartitionKeyGenerators(List<String> par
       return Collections.emptyList(); // Corresponds to no partition case
     } else {
       return partitionPathFields.stream().map(field -> {
-        Pair<String, CustomAvroKeyGenerator.PartitionKeyType> partitionAndType = getPartitionFieldAndKeyType(field);
-        CustomAvroKeyGenerator.PartitionKeyType keyType = partitionAndType.getRight();
+        Pair<String, Option<CustomAvroKeyGenerator.PartitionKeyType>> partitionAndType = getPartitionFieldAndKeyType(field);
+        if (partitionAndType.getRight().isEmpty()) {
+          throw new HoodieKeyException("Unable to find field names for partition path in proper format");
```

Comment:
We can make this more clear to the user. You should explain what the proper format is key:type,...

Reply:
Addressed